app/src/main/java/de/westnordost/streetcomplete/quests/address/AddAddressStreetForm.kt

```diff
@@ -0,0 +1,179 @@
+            if(isPlaceName) {
+                applyAnswer(PlaceName(names))
+            } else {
+                applyAnswer(StreetName(names))
+            }
+        }
+    }
+
+    override val otherAnswers = listOf(
+            OtherAnswer(R.string.quest_address_street_no_named_streets) { switchToPlaceName() }
+    )
+
+    @Inject
+    internal lateinit var abbreviationsByLocale: AbbreviationsByLocale
+    @Inject
+    internal lateinit var roadNameSuggestionsDao: RoadNameSuggestionsDao
+
+    @Inject internal lateinit var favs: LastPickedValuesStore<String>
+
+    init {
+        Injector.instance.applicationComponent.inject(this)
+        val lastPickedNames = favs.get(javaClass.simpleName)
+        defaultName = if (lastPickedNames.isEmpty()) {""} else {lastPickedNames.first}
+    }
+
+    override fun setupNameAdapter(data: List<LocalizedName>, addLanguageButton: View): AddLocalizedNameAdapter {
+
+        return AddLocalizedNameAdapter(
+                data, activity!!, getPossibleStreetsignLanguages(),
+                abbreviationsByLocale, getRoadNameSuggestions(), addLanguageButton,
+                defaultName
+        )
+    }
+
+    private fun getRoadNameSuggestions(): List<MutableMap<String, String>> {
+        return roadNameSuggestionsDao.getNames(
+                geometryToMajorPoints(elementGeometry),
+                AddAddressStreet.MAX_DIST_FOR_ROAD_NAME_SUGGESTION
+        )
+    }
+
+    private fun geometryToMajorPoints(geometry: ElementGeometry): List<LatLon> {
+        return when(geometry) {
+            is ElementPolylinesGeometry -> {
+                val polyline = geometry.polylines.first()
+                listOf(polyline.first(), polyline.last())
+            }
+            is ElementPolygonsGeometry -> {
+                // return center and one of nodes from the way constructing area
+                listOf(geometry.center, geometry.polygons.first().last())
+            }
+            is ElementPointGeometry -> {
+                listOf(geometry.center)
+            }
+        }
+
+
+    }
+
+    override fun isRejectingClose() : Boolean {
+        // if the form is complete, we will reject close, unless it is still has only the default name
+        if (isFormComplete()) {
+            if (adapter.localizedNames.size == 1 && adapter.localizedNames.first().name == defaultName) {
+                return false
+            } else {
+                return true
+            }
+        } else {
+            return false
+        }
+    }
+
+    private fun switchToPlaceName() {
+        AlertDialog.Builder(activity!!)
+                .setTitle(R.string.quest_address_street_noStreet_confirmation_title)
+                .setPositiveButton(R.string.quest_address_street_noStreet_confirmation_positive) {_, _ -> switchToPlaceNameLayout() } //
+                .setNegativeButton(R.string.quest_generic_confirmation_no, null)
+                .show()
+    }
```

Comment:
Confirmation dialogs are only necessary if okay'ing it would lead to the quest being answered. In the housenumber quest, there is no confirmation for "no house number but a name", because it just switches the layout.

Reply:
Fixed in https://github.com/westnordost/StreetComplete/pull/1782/commits/d7f9933ddf9577973fb30a3665ef2e220999b692

Reply:
Ok, I have made it go straight to the layout change in https://github.com/westnordost/StreetComplete/pull/1782/commits/5fd5672363dcde8ead5cff8496882807deec68ac

Reply:
Removed in https://github.com/westnordost/StreetComplete/pull/1782/commits/5fd5672363dcde8ead5cff8496882807deec68ac